lib/puppet/util/puppetdb_validator.rb

```diff
@@ -25,16 +19,17 @@ def attempt_connection
     # All that we care about is that we are able to connect successfully via
     # http(s), so here we're simpling hitting a somewhat arbitrary low-impact URL
     # on the puppetdb server.
-    conn = Puppet::Network::HttpPool.http_instance(puppetdb_server, puppetdb_port, use_ssl)
+    conn = Puppet.runtime[:http]
 
-    response = conn.get(test_path, test_headers)
-    unless response.is_a?(Net::HTTPSuccess)
-      Puppet.notice "Unable to connect to puppetdb server (http#{use_ssl ? 's' : ''}://#{puppetdb_server}:#{puppetdb_port}): [#{response.code}] #{response.msg}"
+    response = conn.get(test_uri, headers: test_headers)
+    if response.is_a?(Puppet::HTTP::ResponseNetHTTP) && response.success?
```

Comment:
All response implementations implement `success?`, so you can drop the `is_a?` check.

Reply:
fixed